### PR TITLE
BC-7664 Rename old stores

### PR DESCRIPTION
**Pull request by wolfganggreschus**
# Short Description

In order to make the work of implementing the room concept easier, the term "room" is largely removed from the current application in the front end, as this usually refers to courses.

## Links to Ticket and related Pull-Requests

BC-7664
## Checklist before merging

- [ ] QA: In addition to review, the code has been manually tested (if manual testing is possible)
- [ ] PO: Any deviation from requirements was agreed with Product-Owner / ticket author / support-team
- [ ] DEV: Every new component is implemented having accessibility in mind (e.g. aria-label, role property)

> Notice: Please keep this Pull-Request as a Draft (or add WIP label), until it is ready to be reviewed

### CODE REVIEW

File: src/components/rooms/CourseBaseCard.unit.ts [hunk 1]
```diff
@@ -3,7 +3,7 @@ import {
 	createTestingI18n,
 	createTestingVuetify,
 } from "@@/tests/test-utils/setup";
-import RoomBaseCard from "./RoomBaseCard.vue";
+import RoomBaseCard from "./CourseBaseCard.vue";
```
Comment (MartinSchuhmacher) [inline src/components/rooms/CourseBaseCard.unit.ts:6]: RoomBaseCard

File: src/components/rooms/CourseExternalToolCard.unit.ts [hunk 2]
```diff
@@ -11,10 +11,10 @@ import {
 	useExternalToolLaunchState,
 } from "@data-external-tool";
 import { createMock, DeepMocked } from "@golevelup/ts-jest";
-import { RoomDotMenu } from "@ui-room-details";
+import { CourseDotMenu } from "@ui-course-details";
 import { flushPromises, mount } from "@vue/test-utils";
 import { nextTick } from "vue";
-import RoomExternalToolCard from "./RoomExternalToolCard.vue";
+import RoomExternalToolCard from "./CourseExternalToolCard.vue";
```
Comment (MartinSchuhmacher) [inline src/components/rooms/CourseExternalToolCard.unit.ts:17]: RoomExternalToolCard

File: src/components/rooms/CourseExternalToolCard.vue [hunk 3]
```diff
@@ -56,10 +56,10 @@ import {
 } from "@data-external-tool";
 import { mdiPencilOutline, mdiTrashCanOutline } from "@mdi/js";
 import { InfoChip, WarningChip } from "@ui-chip";
-import { RoomDotMenu } from "@ui-room-details";
+import { CourseDotMenu } from "@ui-course-details";
 import { computed, ComputedRef, PropType, watch } from "vue";
 import { useI18n } from "vue-i18n";
-import RoomBaseCard from "./RoomBaseCard.vue";
+import RoomBaseCard from "./CourseBaseCard.vue";
```
Comment (MartinSchuhmacher) [inline src/components/rooms/CourseExternalToolCard.vue:62]: RoomBaseCard

**Review by MartinSchuhmacher — CHANGES_REQUESTED** [review]
* room-details/index.ts not needed anymore?

File: src/pages/courses/course-overview.unit.js [hunk 4]
```diff
@@ -439,8 +438,8 @@ describe("@/pages/RoomOverview", () => {
 			},
 		});
 
-		const roomModal = wrapper.findComponent(RoomModal);
-		roomModal.vm.$emit(
+		const CourseModal = wrapper.findComponent({ name: "course-modal" });
```
[the remark this answers is not in the document]
Comment (wolfganggreschus) [reply src/pages/courses/course-overview.unit.js]: wrapper.findComponent(CourseModal); -> doesn't work here. Don't know why...

File: src/store/types/course.ts [hunk 5]
```diff
@@ -44,8 +44,8 @@ export enum CourseFeatures {
 	VIDEOCONFERENCE = "videoconference",
 }
 
-export type RoomData = {
-	roomId: string;
+export type CourseData = {
+	courseId: string;
```
Comment (virgilchiriac) [inline src/store/types/course.ts:48]: todo check if is this accurate

File: src/pages/courses/tools/CourseExternalToolsOverview.vue [hunk 6]
```diff
@@ -27,15 +27,15 @@
 			indeterminate
 		/>
 
-		<room-video-conference-section
+		<course-video-conference-section
 			v-if="isVideoConferenceAvailable"
 			class="mb-4"
-			:room-id="roomId"
+			:room-id="courseId"
```
Comment (virgilchiriac) [inline src/pages/courses/tools/CourseExternalToolsOverview.vue:33]: is this really so? 
should we not see something like :course-id="courseId" or :course-id="roomId"

**Comment by wolfganggreschus** [comment]
Replaced by new PR https://github.com/hpi-schul-cloud/nuxt-client/pull/3361

**Pull request closed without merging**